Fix unexpected smart pointer static analyzer warnings in WebCore and WebKit

rniwa opened the pull request:
#### d8110645bb86fc8679dcf52e4d41c7548bc102a1
<pre>
Fix unexpected smart pointer static analyzer warnings in WebCore and WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=279920">https://bugs.webkit.org/show_bug.cgi?id=279920</a>

Reviewed by Geoffrey Garen.

Fix unexpected smart pointer clang static analyzer warnings in WebCore and WebKit.

* Source/WebCore/css/CSSFontFeatureValue.h:
* Source/WebCore/css/CSSFontStyleWithAngleValue.h:
* Source/WebCore/css/CSSFontVariationValue.h:
* Source/WebCore/style/StyleResolveForFont.cpp:
(WebCore::Style::fontStyleFromCSSValueDeprecated):
(WebCore::Style::fontStyleFromCSSValue):
(WebCore::Style::fontFeatureSettingsFromCSSValue):
(WebCore::Style::fontVariationSettingsFromCSSValue):
(WebCore::Style::fontSizeAdjustFromCSSValue):
* Source/WebKit/GPUProcess/ShapeDetection/RemoteTextDetector.cpp:
(WebKit::RemoteTextDetector::sharedPreferencesForWebProcess const):
(WebKit::RemoteTextDetector::protectedBacking const):
(WebKit::RemoteTextDetector::protectedBackend const):
(WebKit::RemoteTextDetector::protectedBacking): Deleted.
(WebKit::RemoteTextDetector::protectedBackend): Deleted.
* Source/WebKit/GPUProcess/ShapeDetection/RemoteTextDetector.h:
(WebKit::RemoteTextDetector::backing const):
(WebKit::RemoteTextDetector::backing): Deleted.
* Source/WebKit/Shared/Cocoa/WKNSArray.mm:
(-[WKNSArray objectAtIndex:]):
(-[WKNSArray _protectedArray]):
* Source/WebKit/UIProcess/ResponsivenessTimer.cpp:
(WebKit::ResponsivenessTimer::protectedClient const):
(WebKit::ResponsivenessTimer::mayBecomeUnresponsive const):
* Source/WebKit/UIProcess/ResponsivenessTimer.h:

Canonical link: <a href="https://commits.webkit.org/283889@main">https://commits.webkit.org/283889@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/60516a61292fffe1e996972d9926c1ef15ea22e9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67673 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/47055 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/20312 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71727 "Built successfully") | [⏳ 🛠 wincairo ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/69791 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54854 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18617 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/54166 "Passed tests") | [⏳ 🧪 wincairo-tests ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70740 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/43157 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58524 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/34634 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39831 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15929 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/17172 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61794 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/16271 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/73486 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11636 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15575 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/61647 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11671 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58592 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/61692 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9492 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/3113 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10289 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42858 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/43936 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/45125 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43677 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

RupinMittal reviewed (commented):
Looks good to me! You'll have to get the approval from someone else since I'm not a reviewer yet.

geoffreygaren reviewed (approved):
r=me

rniwa committed:
Fix unexpected smart pointer static analyzer warnings in WebCore and WebKit
https://bugs.webkit.org/show_bug.cgi?id=279920

Reviewed by Geoffrey Garen.

Fix unexpected smart pointer clang static analyzer warnings in WebCore and WebKit.

* Source/WebCore/css/CSSFontFeatureValue.h:
* Source/WebCore/css/CSSFontStyleWithAngleValue.h:
* Source/WebCore/css/CSSFontVariationValue.h:
* Source/WebCore/style/StyleResolveForFont.cpp:
(WebCore::Style::fontStyleFromCSSValueDeprecated):
(WebCore::Style::fontStyleFromCSSValue):
(WebCore::Style::fontFeatureSettingsFromCSSValue):
(WebCore::Style::fontVariationSettingsFromCSSValue):
(WebCore::Style::fontSizeAdjustFromCSSValue):
* Source/WebKit/GPUProcess/ShapeDetection/RemoteTextDetector.cpp:
(WebKit::RemoteTextDetector::sharedPreferencesForWebProcess const):
(WebKit::RemoteTextDetector::protectedBacking const):
(WebKit::RemoteTextDetector::protectedBackend const):
(WebKit::RemoteTextDetector::protectedBacking): Deleted.
(WebKit::RemoteTextDetector::protectedBackend): Deleted.
* Source/WebKit/GPUProcess/ShapeDetection/RemoteTextDetector.h:
(WebKit::RemoteTextDetector::backing const):
(WebKit::RemoteTextDetector::backing): Deleted.
* Source/WebKit/Shared/Cocoa/WKNSArray.mm:
(-[WKNSArray objectAtIndex:]):
(-[WKNSArray _protectedArray]):
* Source/WebKit/UIProcess/ResponsivenessTimer.cpp:
(WebKit::ResponsivenessTimer::protectedClient const):
(WebKit::ResponsivenessTimer::mayBecomeUnresponsive const):
* Source/WebKit/UIProcess/ResponsivenessTimer.h:

Canonical link: https://commits.webkit.org/283889@main
